src/splinterdb.c

```diff
@@ -437,13 +437,15 @@ splinterdb_create_or_open(const splinterdb_config *kvs_cfg,      // IN
       goto deinit_kvhandle;
    }
 
-   uint8 num_bg_threads[NUM_TASK_TYPES] = {0}; // no bg threads
+   uint8 *num_bg_threads = (uint8 *)kvs_cfg->num_bg_threads;
+   bool   use_bg_threads = ((num_bg_threads[TASK_TYPE_NORMAL] != 0)
+                          || (num_bg_threads[TASK_TYPE_MEMTABLE] != 0));
 
    status = task_system_create(kvs->heap_id,
                                &kvs->io_handle,
                                &kvs->task_sys,
                                TRUE,
-                               FALSE,
+                               use_bg_threads,
```

Comment:
Can we just delete this flag and have task.c infer it from the `num_bg_threads` arg?

Reply:
Done, @rtjohnso . Add added unit-tests to cross-check that we are handling all cases correctly.